### PR TITLE
Remove public API usage of ComponentSlotStyle and redefine type internally

### PR DESCRIPTION
**Pull request by JamesBurnside**
# What
Redefine `ComponentSlotStyle` with interoperable type from fluentui

This works because IRawStyle allows any `Record<string, any>` but has defined types for common CSS property (of which ComponentSlotStyle also supports) -- so is just a better version of `Record<string, any>` for styles properties.

# Why
We are removing usage of fluent n-star (see #2988) to support react 18 (see #1900) and we need to not rely on this type.

# How Tested
Checked for interopability:

image
image

### CODE REVIEW

**Review by emlynmac — APPROVED** [review]
Great step forward to react 9!

File: packages/react-components/src/types/ComponentSlotStyle.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { IRawStyle } from '@fluentui/react';
+
+/**
```
Comment (PorterNan) [inline packages/react-components/src/types/ComponentSlotStyle.ts:6]: Would this be better to put this one in northstar wrapper?

**Review by PorterNan — APPROVED** [review]

File: packages/react-components/src/types/ComponentSlotStyle.ts (hunk 1 above)
Comment (JamesBurnside) [reply packages/react-components/src/types/ComponentSlotStyle.ts:6]: I did consider that but I'm actually against that as we shouldn't ever have publicly exported a fluent northstar interface and should have defined our own interface/used regular fluents in this package.

Comment (JamesBurnside) [reply packages/react-components/src/types/ComponentSlotStyle.ts:6]: (It also makes the bundling more difficult if we have a public API coming out of the northstar wrapper the way we rollup the api)

**Pull request merged**